packages/wallets/src/metamask/metamask.page.ts

```diff
@@ -127,19 +120,47 @@ export class MetamaskPage implements WalletPage {
     chainId: number,
     tokenSymbol: string,
     blockExplorer = '',
+    isClosePage = false,
   ) {
     await test.step(`Add new network "${networkName}"`, async () => {
-      await this.settingsPage.openSettings();
-      await this.settingsPage.networksTabButton.click();
-      await this.settingsPage.addNetworkManually(
-        networkName,
-        networkUrl,
-        chainId,
-        tokenSymbol,
-        blockExplorer,
-      );
-      await this.popoverElements.switchToButton.click();
+      await this.navigate();
+      await this.header.networkListButton.click();
+      const networkListText = await this.networkList.getNetworkListText();
+      if (networkListText.includes(networkName)) {
+        await this.networkList.clickToNetworkItemButton(networkName);
+      } else {
+        await this.networkList.networkDisplayCloseBtn.click();
+        await this.settingsPage.openSettings();
+        await this.settingsPage.networksTabButton.click();
+        await this.settingsPage.addNetworkManually(
```

Comment:
need to merge develop and reuse new flow since network add flow has changed

Reply:
done.
+ move some functions from new `networkSetting` class to `networkList` class